src/tests/integration/integration-test.py

```diff
@@ -34,28 +36,37 @@ def setup(cls, config_path, openapi_path):
     def cleanup(cls):
         cls.session.close()
 
-    # Test case: GET /pets
-    def test_get_all_pets(self, endpoint='/pets'):
-        nullable_fields = ['owner']
-        utils.test_endpoint(self, endpoint, 'PetResource', 200,
-                            nullable_fields=nullable_fields)
-
-    # Test case: GET /pets with species filter
-    def test_get_pets_with_filter(self, endpoint='/pets'):
-        testing_species = ['dog', 'CAT', 'tUrTlE']
-
-        for species in testing_species:
-            params = {'species': species}
-            response = utils.test_endpoint(self, endpoint, 'PetResource', 200,
-                                           query_params=params)
-
+    # Test GET /directory
+    def test_get_directory(self, endpoint='/directory'):
+        tests = {}
+        test_cases = self.test_cases
+        for param in test_cases:
+            if param != 'id':
```

Comment:
I'm a little confused on whats going on here. Are you just trying to make a copy of `test_cases` without the `id` key? something like [this](https://stackoverflow.com/a/31434043) could simplify it.

Reply:
I was getting keyErrors when trying to access things from self.test_cases, so the idea there was to copy the json object into a locally stored dict. I figured out what I was doing wrong, and I updated the logic to just use the original self.test_cases dict and ignore the 'id' when calling the test_query_params helper